src/client/dfuse/ops/listxattr.c

```diff
@@ -30,17 +32,26 @@ dfuse_cb_listxattr(fuse_req_t req, struct dfuse_inode_entry *inode,
 
 	D_ALLOC(value, out_size);
 	if (!value)
-		D_GOTO(err, rc = ENOMEM);
+		D_GOTO(free, rc = ENOMEM);
 
-	rc = dfs_listxattr(inode->ie_dfs->dfs_ns, inode->ie_obj, value,
-			   &out_size);
+	if (inode->ie_root) {
+		out_size -= 10;
+	}
+
+	rc = dfs_listxattr(inode->ie_dfs->dfs_ns, inode->ie_obj, value, &out_size);
 	if (rc != 0)
-		D_GOTO(err, rc);
+		D_GOTO(free, rc);
+
+	if (inode->ie_root) {
+		sprintf(&value[out_size], "user.daos");
+		out_size += 10;
+	}
```

Comment:
It's a buffer of known lengths, which contains multiple NULL terminated strings.  The additional space is 9 for the string and one for the extra termination needed.

I assume that dfs_listxattr includes a NULL terminator on the end of the last string but can see that it might not be required to, however testing would have picked up any issue here.

Reply:
You're right. I misread this. (too early in the morning for me :) )